src/ostree/ot-builtin-commit.c

```diff
@@ -196,6 +198,13 @@ commit_filter (OstreeRepo         *self,
       g_hash_table_remove (mode_adds, path);
     }
 
+  if (opt_canonical_permissions &&
+      g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR)
```

Comment:
Yep, I understand the rationale for the option, but I'd still say it should be on by default for bare-user.  Among other things, every API caller will need to duplicate this logic if it's not.